webapp/src/main/java/org/opentestsystem/rdw/admin/service/impl/DefaultStudentGroupBatchService.java

```diff
@@ -173,10 +175,9 @@ private void archiveFile(final MultipartFile file,
                 return validationFailures;
             }
 
-            final List<String> orderedHeaders = new ArrayList<>(parser.getHeaderMap().size());
-            for (final Map.Entry<String, Integer> entry : parser.getHeaderMap().entrySet()) {
-                orderedHeaders.add(entry.getValue(), entry.getKey());
-            }
```

Comment:
Oops.  I had mistakenly assumed that initializing the list with a capacity allowed me to insert up to that capacity without an index out-of-bounds exception.
Switched to insert-and-sort method below.